### PR TITLE
patch retro_opendir to handle nullptr and empty-string input

### PR DESCRIPTION
**Pull request by hiddenasbestos**
Strengthen retro_opendir.

### CODE REVIEW

File: libretro-common/file/retro_dirent.c [hunk 1]
```diff
@@ -108,9 +108,13 @@ struct RDIR *retro_opendir(const char *name)
 #endif
    struct RDIR *rdir  = (struct RDIR*)calloc(1, sizeof(*rdir));
 
-   if (!rdir)
+   if (!rdir||!name)
       return NULL;
 
+   /* Handle empty string as current dir */
+   if (*name==0)
+      name=".";
```
Comment (i30817) [inline libretro-common/file/retro_dirent.c]: If you're going to do this, doesn't it ned to be null terminated?

Comment (hiddenasbestos) [reply libretro-common/file/retro_dirent.c]: Yeah the double quotes should create two bytes of static data in the EXE containing a full-stop and a terminating zero, and the assignment will then point to it.

**Comment by twinaphex** [comment]
This empty-string input on second thought -

'   /* Handle empty string as current dir */'

I find this to be highly dangerous and inherently non-portable, and it might be best to remove this hack and instead let the caller deal with these kinds of scenarios instead.

**Comment by i30817** [comment]
assert it?

**Comment by hiddenasbestos** [comment]
Okay @twinaphex I'll look into a better patch.

There is a crash that will occur if you pass in "" on _WIN32 builds as it reads from [length-1] so it does need to be strengthened in some regard.

However, what would you suggest the behaviour should be for retro_opendir( "" ) -- return null perhaps?

**Comment by hiddenasbestos** [comment]
@twinaphex something like:

   if (!rdir||!name||(*name==0))
      return NULL;

**Pull request merged**